wp/lib/bap_wp/src/compare.ml

```diff
@@ -35,13 +38,40 @@ let set_to_eqs (env1 : Env.t) (env2 : Env.t) (vars : Var.Set.t) : Constr.t list
     ~f:(fun (eqs, env1, env2) v ->
         let var1, env1 = Env.get_var env1 v in
         let var2, env2 = Env.get_var env2 v in
-        let eq = Bool.mk_eq ctx var1 var2
-                 |> Constr.mk_goal
-                   (Format.sprintf "%s = %s" (Expr.to_string var1) (Expr.to_string var2))
-                 |> Constr.mk_constr
-        in (eq::eqs, env1, env2)
+        if Target.CPU.is_mem v then
+          eqs, env1, env2
+        else
+          let eq = Bool.mk_eq ctx var1 var2
+                   |> Constr.mk_goal
+                     (Format.sprintf "%s = %s" (Expr.to_string var1) (Expr.to_string var2))
+                   |> Constr.mk_constr
+          in (eq::eqs, env1, env2)
       )
 
+(* Adds the hypothesis: [var == init_var]. *)
```

Comment:
Do we have docs on how to use `foo_init` in specs? Do we have examples? Do we have C examples?

Reply:
Nope, but I can definitely write up some documentation on the init vars (in the .mli?) and a little test example.